harness/deepEqual.js

```diff
@@ -6,268 +6,322 @@ description: >
-        return !test(a)
-            ? !test(b) ? UNKNOWN : NOT_EQUAL
-            : !test(b) ? NOT_EQUAL : cacheComparison(a, b, compare, cache);
-    }
-
-    function tryCompareStrictEquality(a, b) {
-        return a === b ? EQUAL : UNKNOWN;
-    }
-
-    function tryCompareTypeOfEquality(a, b) {
-        return typeof a !== typeof b ? NOT_EQUAL : UNKNOWN;
-    }
-
-    function tryCompareToStringTagEquality(a, b) {
-        var aTag = Symbol.toStringTag in a ? a[Symbol.toStringTag] : undefined;
-        var bTag = Symbol.toStringTag in b ? b[Symbol.toStringTag] : undefined;
-        return aTag !== bTag ? NOT_EQUAL : UNKNOWN;
-    }
-
-    function isOptional(value) {
-        return value === undefined
-            || value === null;
-    }
-
-    function compareOptionality(a, b) {
-        return tryCompareStrictEquality(a, b)
-            || NOT_EQUAL;
-    }
-
-    function isPrimitiveEquatable(value) {
-        switch (typeof value) {
-            case 'string':
-            case 'number':
-            case 'bigint':
-            case 'boolean':
-            case 'symbol':
-                return true;
-            default:
-                return isBoxed(value);
-        }
-    }
-
-    function comparePrimitiveEquality(a, b) {
-        if (isBoxed(a)) a = a.valueOf();
-        if (isBoxed(b)) b = b.valueOf();
-        return tryCompareStrictEquality(a, b)
-            || tryCompareTypeOfEquality(a, b)
-            || compareIf(a, b, isNaNEquatable, compareNaNEquality)
-            || NOT_EQUAL;
-    }
-
-    function isNaNEquatable(value) {
-        return typeof value === 'number';
-    }
-
-    function compareNaNEquality(a, b) {
-        return isNaN(a) && isNaN(b) ? EQUAL : NOT_EQUAL;
-    }
-
-    function isObjectEquatable(value) {
-        return typeof value === 'object';
-    }
+assert.deepEqual = function(actual, expected, message) {
+  var format = assert.deepEqual.format;
+  assert(
+    assert.deepEqual._compare(actual, expected),
+    `Expected ${format(actual)} to be structurally equal to ${format(expected)}. ${(message || '')}`
+  );
+};
 
-    function compareObjectEquality(a, b, cache) {
-        if (!cache) cache = new Map();
-        return getCache(cache, a, b)
-            || setCache(cache, a, b, EQUAL) // consider equal for now
-            || cacheComparison(a, b, tryCompareStrictEquality, cache)
-            || cacheComparison(a, b, tryCompareToStringTagEquality, cache)
-            || compareIf(a, b, isValueOfEquatable, compareValueOfEquality)
-            || compareIf(a, b, isToStringEquatable, compareToStringEquality)
-            || compareIf(a, b, isArrayLikeEquatable, compareArrayLikeEquality, cache)
-            || compareIf(a, b, isStructurallyEquatable, compareStructuralEquality, cache)
-            || compareIf(a, b, isIterableEquatable, compareIterableEquality, cache)
-            || cacheComparison(a, b, fail, cache);
-    }
+assert.deepEqual.format = function(value, seen) {
```

Comment:
this is the former `assert._formatValue`, not so exposed anymore and meant to be used here only.